internals/overlord/checkstate/manager.go

```diff
@@ -349,10 +375,75 @@ type CheckInfo struct {
 type CheckStatus string
 
 const (
-	CheckStatusUp   CheckStatus = "up"
-	CheckStatusDown CheckStatus = "down"
+	CheckStatusUp       CheckStatus = "up"
+	CheckStatusDown     CheckStatus = "down"
+	CheckStatusInactive CheckStatus = "inactive"
 )
 
 type checker interface {
 	check(ctx context.Context) error
 }
+
+// StartChecks starts the specified checks by creating a new perform-checks
+// change for each check that is in the current plan and not already running.
+func (m *CheckManager) StartChecks(currentPlan *plan.Plan, checks []string) ([]string, error) {
```

Comment:
I think I'd prefer that these methods fetched the plan via the manager, rather than the caller having to pass in the plan. Can we drop it from the signatures here, and instead have `NewManager` take a reference to and store the `*PlanManager`? That seems reasonable, as the check manager now needs to know about the plan for replanning purposes.